src/main/java/org/gaul/s3proxy/S3ProxyHandler.java

```diff
@@ -2221,6 +2224,10 @@ private void handleCopyPart(HttpServletRequest request,
         }
 
         BlobMetadata blobMetadata = blob.getMetadata();
+        // HTTP GET allow overlong ranges but S3 CopyPart does not
+        if (expectedSize != -1 && blobMetadata.getSize() < expectedSize) {
```

Comment:
Normal GET handling will reject the invalid beginning.  This code only needs to add the additional CopyPart checks.